lib/racecar/message_delivery_error.rb

```diff
@@ -0,0 +1,100 @@
+        EOM
+
+      when :msg_size_too_large # 10
+        <<~EOM
+          Could not deliver message, since it is bigger than either the broker's or Racecar's maximum message size.
+
+          The broker's config option on the topic is called "max.message.bytes" and the broker wide default is "message.max.bytes". The client's is "message.max.bytes". Take extra care to distinguish this from similarly named properties for receiving/consuming messages (i.e. Racecar.config.max_bytes is NOT related).
+
+          Racecar's limit is currently not configurable and uses librdkafka's default of 1 MB (10³ bytes). As of writing, librdkafka will send at least one message regardless of this limit. It is therefore very likely you're hitting the broker's limit and not Racecar's/librdkafka's.
+
+          Upstream documentation:
+          broker per topic: https://docs.confluent.io/platform/current/installation/configuration/topic-configs.html#topicconfigs_max.message.bytes
+          broker default:   https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#brokerconfigs_message.max.bytes
+          client:           https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        EOM
+
+      when :unknown_topic_or_part # 3
+        <<~EOM
+          Could not deliver message, since the targeted topic or partition (#{@report&.partition}) does not exist.
+
+          Check that there are no typos, or that the broker's "auto.create.topics.enable" is enabled. For freshly created topics with auto create enabled, this may appear in the beginning (race condition on creation and publishing).
+
+          Upstream documentation:
+          broker setting: https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#brokerconfigs_auto.create.topics.enable
+          client:         https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#topic-metadata-propagation-for-newly-created-topics
+                          https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#topic-auto-creation
+        EOM
+
+      when :record_list_too_large # 18
+        <<~EOM
+          Tried to deliver more messages in a batch than the broker's segment size.
+
+          Either increase the broker's "log.segment.bytes", or decrease any of the client's related settings "batch.num.messages", "batch.size" or "message.max.bytes". None of these are configurable through Racecar yet, as the defaults should be sufficient and sane.
+
+          Upstream documentation:
+          broker: https://docs.confluent.io/platform/current/installation/configuration/broker-configs.html#brokerconfigs_log.segment.bytes
+          client: https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        EOM
+
+      when :topic_authorization_failed # 29
+        <<~EOM
+          Failed to deliver message because lacking the authorization to write into the topic.
```

Comment:
```suggestion
          Failed to deliver message because of insufficient authorization to write into the topic (#{@report&.topic}).
```

Is `topic` available on `@report`?

Reply:
People overwriting `produce` are already breaking the implicit contract of Racecar – that's not something you should do. If you're up for it, I think it would make sense to look into a Racecar wrapper for the delivery handles that kept that extra data. Doesn't have to be in this PR, though.